File: 2427-first-letter-to-appear-twice/first-letter-to-appear-twice.java
class Solution {
    public char repeatedCharacter(String s) {
        HashMap<Character,Integer> mp=new HashMap<>();

        for (int i = 0; i < s.length(); i++){
            char it=s.charAt(i);
            if(mp.containsKey(it)){
                return it;
            }
            else{
                mp.put(it, mp.getOrDefault(it, 0) + 1);
            }
        }

        return ' ';
    }
}

File: 2427-first-letter-to-appear-twice/first-letter-to-appear-twice.cpp
class Solution {
public:
    char repeatedCharacter(string s) {
        unordered_map<char,int> mp;

        for(char& it:s){
            if(mp.find(it)!=mp.end()){
                return it;
            }
            else{
                mp[it]++;
            }
        }
        return ' ';
    }
};